steps/bolt-step-run/spec.schema.json

```diff
@@ -0,0 +1,119 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "type": {
+      "type": "string",
+      "description": "The type of Bolt action to perform, one of `task` or `plan`."
+    },
+    "name": {
+      "type": "string",
+      "description": "The name of the Bolt task to plan to run."
+    },
+    "parameters": {
+      "type": "object",
+      "description": "The task- or plan-specific parameter data to use for this run.",
+      "additionalProperties": {
+        "type": "string"
+      }
+    },
+    "projectDir": {
+      "type": "string",
+      "description": "When the `git` setting is specified, the relative directory within the git repository to use as the Bolt project.",
+      "default": "."
+    },
+    "modulePaths": {
+      "type": "array",
+      "items": {
+        "type": "string"
+      },
+      "description": "A list of additional directories outside of the project directory to add to the Bolt module path."
+    },
+    "installModules": {
+      "type": "boolean",
+      "description": "Download and install modules listed in a `Puppetfile`.",
+      "default": false
+    },
+    "transport": {
+      "type": "object",
+      "description": "The default transport to use when connecting to the target nodes.",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The transport type, one of `ssh` or `winrm`"
+        },
+        "user": {
+          "type": "string",
+          "description": "The username to use to connect."
+        },
+        "password": {
+          "type": "string",
+          "description": "The password to use to connect."
+        },
+        "privateKey": {
+          "type": "string",
+          "description": "The name of a key in the `credentials` setting to use as the default SSH private key."
+        },
+        "useSSL": {
+          "type": "boolean",
+          "description": "If using the WinRM transport, whether to use SSL to connect.",
+          "default": true
+        },
+        "verifyHost": {
+          "type": "boolean",
+          "description": "Whether to verify host integrity when connecting to remote nodes.",
+          "default": true
+        }
+      }
+    },
+    "targets": {
+      "type": ["string", "array"],
+      "items": { "type": "string" },
```

Comment:
This might make more sense as a `oneOf: [{"type": "string"}, {"type": "array", ...}]` but I'd say we make a note and revisit it later.

Reply:
I actually had that in the commit above, if you want to look at it.